woocommerce/woocommerce
# [e2e] Add support for WPCOM site

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Part of pdV5qK-wB-p2

1. Add support to run e2e tests against WPCOM website
   - Following the changes in #49017  I added an option to run tests against WPCOM website
2. Updated the default Pressable website to run e2e tests. 



### How to test the changes in this Pull Request:

0. Find `E2E_ENV_KEY` in the secret store and export it as an environment variable. 
1. Checkout this branch 
2. Run `pnpm test:e2e:with-env default-pressable`
   - File tests/e2e-pw/envs/default-pressable/.env.enc is decrypted into tests/e2e-pw/.env
3. Run `pnpm test:e2e:with-env default-wpcom`
   - File tests/e2e-pw/envs/default-wpcom/.env.enc is decrypted into tests/e2e-pw/.env

Admin and customer should log in successfully and tests should start. 
NOTE: At the moment, test "Load the home page" is failing for WPCOM, but that will be the part of another investigation.

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add support to run e2e tests against WPCOM website. 

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
